plugins/entity-validation/src/utils/utils.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Entity, stringifyLocationRef } from '@backstage/catalog-model';
+import lodash from 'lodash';
+import yaml from 'yaml';
+import { LocationSpec } from '@backstage/plugin-catalog-common';
+import {
+  CatalogProcessorResult,
+  processingResult,
+} from '@backstage/plugin-catalog-node';
```

Comment:
Hm. Frontend packages should not import backend (Node) packages, and vice versa. They are built completely differently, and even when it at times does work technically, you want to avoid polluting your packages with this extra weight of Node stuff getting pulled in.

I'd suggest rewriting the function below to not leverage these things at all; you can just copy over the relevant result types and perhaps even modify to suit your needs. Something as basic as `type Result = { entity: ... } | { error: ... }` or so.

Remember to remove the dependency from package.json then, too.

Reply:
Good catch. I had that issue with the function to parse the json to a list of entities, but there I was importing a `-backend` package. As soon as I got time for it, I will do so